src/function_lua.c

```diff
@@ -137,13 +163,78 @@ static void luaEngineFreeFunction(void *engine_ctx, void *compiled_function) {
     zfree(f_ctx);
 }
 
+static int luaRegisterFunction(lua_State *lua) {
+    int argc = lua_gettop(lua);
+    if (argc < 2 || argc > 3) {
+        luaPushError(lua, "wrong number of arguments to redis.register_function");
+        return luaRaiseError(lua);
+    }
+    loadCtx *load_ctx = luaGetFromRegistry(lua, REGISTRY_LOAD_CTX_NAME);
+    if (!load_ctx) {
+        luaPushError(lua, "redis.register_function can only be called on FUNCTION LOAD command");
+        return luaRaiseError(lua);
+    }
+
+    if (!lua_isstring(lua, 1)) {
+        luaPushError(lua, "first argument to redis.register_function must be a string");
+        return luaRaiseError(lua);
+    }
+
+    if (!lua_isfunction(lua, 2)) {
+        luaPushError(lua, "second argument to redis.register_function must be a function");
+        return luaRaiseError(lua);
+    }
+
+    if (argc == 3 && !lua_isstring(lua, 3)) {
+        luaPushError(lua, "last argument to redis.register_function must be a string");
```

Comment:
who knows if we'll add more args in the future..
```suggestion
        luaPushError(lua, "third argument to redis.register_function must be a string");
```